icu4c/source/common/rbbi.cpp

```diff
@@ -194,10 +190,35 @@ RuleBasedBreakIterator::RuleBasedBreakIterator( const UnicodeString  &rules,
 //                           of rules.
 //-------------------------------------------------------------------------------
 RuleBasedBreakIterator::RuleBasedBreakIterator()
- : fSCharIter(UnicodeString())
+ : RuleBasedBreakIterator(fErrorCode)
 {
-    UErrorCode status = U_ZERO_ERROR;
-    init(status);
+}
+
+/**
+ * Simple Constructor with an error code.
+ * Handles common initialization for all other constructors.
+ */
+RuleBasedBreakIterator::RuleBasedBreakIterator(UErrorCode &status) {
+    utext_openUChars(&fText, nullptr, 0, &status);
+    LocalPointer<DictionaryCache> lpDictionaryCache(new DictionaryCache(this, status), status);
+    LocalPointer<BreakCache> lpBreakCache(new BreakCache(this, status), status);
+    if (U_FAILURE(status)) {
+        fErrorCode = status;
```

Comment:
This data member is never read. Isn't the parameter sufficient for the status?

Reply:
Never mind. This is mentioned later on as being used with copyErrorTo in a future pull request.